### PR TITLE
Move `join` to new subctl implementation

### PR DESCRIPTION
**Pull request by Jaanki**
Signed-Off-By: Janki Chhatbar <jchhatba@redhat.com>

1. Move functions from `join.go` to `helpers.go`
2. Make `BrokerDetails` and `ClusterDetails` subset of `SubmarinerSpec` and `ServiceDiscoverySpec`
3. Move `image*` functions to `internal/image` package

### CODE REVIEW

File: api/submariner/v1alpha1/servicediscovery_types.go [hunk 1]
```diff
@@ -30,21 +30,9 @@ import (
...
-	Version                  string               `json:"version,omitempty"`
-	Debug                    bool                 `json:"debug"`
-	GlobalnetEnabled         bool                 `json:"globalnetEnabled,omitempty"`
-	BrokerK8sInsecure        bool                 `json:"brokerK8sInsecure,omitempty"`
-	CoreDNSCustomConfig      *CoreDNSCustomConfig `json:"coreDNSCustomConfig,omitempty"`
-	// +listType=set
-	CustomDomains  []string          `json:"customDomains,omitempty"`
-	ImageOverrides map[string]string `json:"imageOverrides,omitempty"`
+	BrokerDetails    BrokerDetails  `json:"broker_info"`
+	ClusterDetails   ClusterDetails `json:"cluster_info"`
```
Comment (skitt) [inline api/submariner/v1alpha1/servicediscovery_types.go]: This is a breaking change. You can share struct components but only by “inheriting” the struct. Any changes made here shouldn’t result in changes to the CRD YAMLs.

File: api/submariner/v1alpha1/submariner_types.go [hunk 2]
```diff
@@ -31,38 +31,46 @@ import (
...
+
 // SubmarinerSpec defines the desired state of Submariner
 // +k8s:openapi-gen=true
 type SubmarinerSpec struct {
 	Broker                   string               `json:"broker"`
-	BrokerK8sApiServer       string               `json:"brokerK8sApiServer"`
-	BrokerK8sApiServerToken  string               `json:"brokerK8sApiServerToken"`
-	BrokerK8sCA              string               `json:"brokerK8sCA"`
-	BrokerK8sRemoteNamespace string               `json:"brokerK8sRemoteNamespace"`
+	BrokerDetails            BrokerDetails        `json:"broker_details"`
```
Comment (skitt) [inline api/submariner/v1alpha1/submariner_types.go]: Same as above, this is a breaking change, as is the `ClusterDetails` field below.

**Review by skitt — CHANGES_REQUESTED** [review]

**Comment by tpantelis** [comment]
I assume this is superseded by https://github.com/submariner-io/submariner-operator/pull/1726?

**Comment by Jaanki** [comment]
Closed in favor of https://github.com/submariner-io/submariner-operator/pull/1726

**Pull request closed without merging**